src/SOS/SOS.Hosting/SOSHost.cs

```diff
@@ -45,72 +56,62 @@ public sealed class SOSHost : IDisposable
         private readonly SOSLibrary _sosLibrary;
 #pragma warning restore
 
-        private readonly IntPtr _interface;
+        private readonly IntPtr _client;
         private readonly ulong _ignoreAddressBitsMask;
-        private bool _disposed;
+        private readonly bool _releaseClient;
 
         /// <summary>
         /// Create an instance of the hosting class. Has the lifetime of the target.
         /// </summary>
-        public SOSHost(ITarget target, IMemoryService memoryService)
+        public SOSHost(ITarget target, IMemoryService memoryService, [ServiceImport(Optional = true)] INativeClient client)
         {
-            Target = target ?? throw new DiagnosticsException("No target");
+            Target = target;
             MemoryService = memoryService;
             _ignoreAddressBitsMask = memoryService.SignExtensionMask();
 
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            // If running under a native debugger, use the client instance supplied by the debugger for commands
+            if (client != null)
             {
-                DebugClient debugClient = new(this);
-                _interface = debugClient.IDebugClient;
+                _client = client.Client;
             }
             else
             {
-                LLDBServices lldbServices = new(this);
-                _interface = lldbServices.ILLDBServices;
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    DebugClient debugClient = new(this);
+                    _client = debugClient.IDebugClient;
+                }
+                else
+                {
+                    LLDBServices lldbServices = new(this);
+                    _client = lldbServices.ILLDBServices;
+                }
+                _releaseClient = true;
             }
         }
 
         void IDisposable.Dispose()
         {
-            Trace.TraceInformation($"SOSHost.Dispose {_disposed}");
-            if (!_disposed)
+            Trace.TraceInformation($"SOSHost.Dispose");
+            if (_releaseClient)
```

Comment:
I wanted to keep this logic simple and not have multiple conditions.  All the other disposable services don't have a disposed flag and the DI/service infrastructure guarantees that it will never get called more than once.  Commands and other services never directly dispose services.